unblob/logging.py

```diff
@@ -100,14 +106,40 @@ def configure_logger(verbosity_level: int, extract_root: Path):
         structlog.processors.UnicodeDecoder(),
         structlog.processors.StackInfoRenderer(),
         structlog.processors.format_exc_info,
-        structlog.dev.ConsoleRenderer(colors=sys.stdout.isatty()),
     ]
 
     structlog.configure(
         wrapper_class=structlog.make_filtering_bound_logger(log_level),
-        processors=processors,
+        processors=shared_processors + processors,
+        logger_factory=structlog.stdlib.LoggerFactory(),
+    )
+
+    formatter = structlog.stdlib.ProcessorFormatter(
+        foreign_pre_chain=shared_processors,
+        processors=[
+            structlog.stdlib.ProcessorFormatter.remove_processors_meta,
+            structlog.dev.ConsoleRenderer(colors=sys.stdout.isatty()),
+        ],
+    )
+
+    file_formatter = structlog.stdlib.ProcessorFormatter(
+        foreign_pre_chain=shared_processors,
+        processors=[
+            structlog.stdlib.ProcessorFormatter.remove_processors_meta,
+            structlog.dev.ConsoleRenderer(colors=False),
+        ],
     )
 
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setFormatter(formatter)
+    file_handler = logging.handlers.WatchedFileHandler("unblob.log")  # type: ignore
```

Comment:
An open question is whether we should always use the same file or create a temp file and provide the name as part of console output.

Reply:
maybe the user can specify the filename and we can default to a filename similar to the input filename? This way just running it would not overwrite existing file, but for scripting they can specify and do not need to guess it?

Reply:
Log files based on the input is a good idea, and it is tolerable beside the extraction directory, but I also feel it is nice to have for only some and only in a few cases and not in general. It does hurt a bit over `unblob.log`: it is somewhat harder to properly script against (thinking about shell escapes here).

Making it configurable is time and code without clear use in sight, so why?

I still prefer a single well known "unblob.log".

Reply:
I just added a `--log` option so user can provide an explicit log file. The value defaults to `unblob.log` if it's not provided.

The log file default verbosity is 2 so that it's not bloated with escape sequences and colors coming from `dissect.cstruct` structures dumped in the logs. Verbosity level 3 is only used by contributors during development anyway.

 I had to set the `exception_formatter` explicitly to `structlog.dev.plain_traceback`. Otherwise `rich` takes over and formats exceptions with displays and colors which severely impact performance.